Lib/Metaiterators.hpp

```diff
@@ -369,7 +369,7 @@ class FilteredIterator
   DECL_ELEMENT_TYPE(ELEMENT_TYPE(Inner));
 
   FilteredIterator(Inner inn, Functor func)
-  : _func(func), _inn(inn), _nextStored(false) {}
+  : _func(func), _inn(inn), _next(), _nextStored(false) {}
   bool hasNext()
```

Comment:
This one looks innocent, but is actually pernicious. ELEMENT_TYPE is both POD (usually raw pointers) or user-defined types in Vampire. In the POD case this change just adds the same zero-initialise as elsewhere.

If it's a user-defined type, this change has no effect. However, I notice that this means any use of the filter class will incur whatever the default constructor does, then throw it away later. We probably don't want (or intend?) to do this, but these are the current semantics of `FilteredIterator` so I don't change it here.

Reply:
In other word, the new version is just explicitly saying what was implicitly happening anyway, but we know suspect, it's not such a great idea for non PODs, right?

Reply:
Exactly that!